src/v1/connectors/kube.js

```diff
@@ -157,10 +157,16 @@ export default class KubeConnector {
     return this.doRequest(this.getDefaults('DELETE', path, { json: jsonBody }), opts);
   }
 
-  patch(path = '', opts = {}) {
-    const headers = {
-      'Content-Type': 'application/json-patch+json',
-    };
+  // The Kubernetes API server will not recursively create nested objects for a JSON patch input.
+  // We need to use merge-patch here
+  patch(path = '', opts = {}, mergePatch) {
```

Comment:
I think you can just use opts to override the header

Reply:
Ok

Reply:
```  doRequest(defaults, opts) {
    const options = _.merge(defaults, opts);
    this.checkUrl(options.url);
    return this.http(options).then((res) => res.body);
  }
```
`doRequest` accepts `opts`. You can just put header in it and it will merge.